x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ApiKeyService.java

```diff
@@ -155,4 +194,128 @@ private void ensureEnabled() {
             throw new IllegalStateException("tokens are not enabled");
         }
     }
+
+    // pkg-scope for testing
+    void checkIfRoleIsASubsetAndModifyRoleDescriptorsIfRequiredToMakeItASubset(final List<RoleDescriptor> requestRoleDescriptors,
+            final Authentication authentication, ActionListener<List<RoleDescriptor>> listener) throws IOException {
+
+        authorizationService.roles(requestRoleDescriptors, ActionListener.wrap(thisRole -> {
```

Comment:
we shouldn't use the authorization service here; lets use the composite roles store. I understand why you used the authorization service, since we need to get the roles including the anonymous user role names. Maybe we make that a static method and instead have a reference to the anonymous user here so we can use the static method